src/cid.ts

```diff
@@ -270,17 +207,16 @@ export class CID {
       return new CID(
         version,
         code,
-        /** @type {API.MultihashDigest<Alg>} */ (multihash),
-        bytes || encodeCID(version, code, multihash.bytes)
+        (multihash as API.MultihashDigest<Alg>),
+        bytes ?? encodeCID(version, code, multihash.bytes)
       )
     } else if (value[cidSymbol] === true) {
       // If value is a CID from older implementation that used to be tagged via
       // symbol we still rebase it to the this `CID` implementation by
       // delegating that to a constructor.
       const { version, multihash, code } = value
       const digest =
```

Comment:
similar to above, compact the multiline now it's not necessary, remove `(` `)` if unnecessary